projects/resettlement-passport-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/entity/Appointment.kt

```diff
@@ -144,4 +179,43 @@ interface AppointmentRepository : JpaRepository<Appointment, Long> {
     """
     )
     fun findAppointmentsFor(crn: String, start: LocalDate, end: LocalDate, pageable: Pageable): Page<Appointment>
+
+    @Query(
+        """
+            select count(c.contact_id)
+            from contact c
+            join r_contact_type ct on c.contact_type_id = ct.contact_type_id
+            where c.offender_id = :personId and ct.attendance_contact = 'Y'
+            and to_char(c.contact_date, 'YYYY-MM-DD') = :date
+            and to_char(c.contact_start_time, 'HH24:MI') < :endTime 
+            and to_char(c.contact_end_time, 'HH24:MI') > :startTime
```

Comment:
Should this be an OR check, to account for any overlap whether it's at the start or end?
```suggestion
            and (to_char(c.contact_start_time, 'HH24:MI') < :endTime or to_char(c.contact_end_time, 'HH24:MI') > :startTime)
```

Reply:
Sorry - ignore me!